pymc3/distributions/logp.py

```diff
@@ -260,6 +267,130 @@ def _logp(
     """
     value_var = rvs_to_values.get(var, var)
     return at.zeros_like(value_var)
+    # raise NotImplementedError(f"Logp cannot be computed for op {op}")
+
+
+@_logp.register(Elemwise)
+def logp_elemwise(op, *args, **kwargs):
+    if hasattr(op, "scalar_op"):
+        return _logp(op.scalar_op, *args, **kwargs)
+    raise NotImplementedError
+
+
+# TODO: Implement DimShuffle logp?
+# @_logp.register(DimShuffle)
+# def logp_dimshuffle(op, var, *args, **kwargs):
+#     if var.owner and len(var.owner.inputs) == 1:
+#         inp = var.owner.inputs[0]
+#         if inp.owner and hasattr(inp.owner, 'op'):
+#             return _logp(inp.owner.op, inp, *args, **kwargs)
+#     raise NotImplementedError
+
+
+def find_rv_branch(inputs):
+    """
+    Helper function to find which input branch(es) contain unregistered random variables
+    """
+    rv_branch = []
+    no_rv_branch = []
+
+    for inp in inputs:
+        res_ancestors = list(walk_model((inp,), walk_past_rvs=True))
+        # unregistered variables do not contain a value_var tag
+        res_unregistered_ancestors = [
+            v
+            for v in res_ancestors
+            if v.owner
+            and isinstance(v.owner.op, RandomVariable)
+            and not getattr(v.tag, "value_var", False)
+        ]
+        if res_unregistered_ancestors:
+            rv_branch.append(inp)
+        else:
+            no_rv_branch.append(inp)
+
+    return rv_branch, no_rv_branch
```

Comment:
This was a hackish way of deciding which branch needs to be walked to find the logpt of a new rv and which can be treated as a constant.

The thing about registered and unregistered rvs is exactly that and the goal was to not double count an independent variable that was created upstream and instead treat it as a constant into this logpt.

```python
with pm.Model() as m:
  a = pm.Normal('a', 0, 1). # registered rv which will be in the model logpt and can be used as a constant downstream
  b = pm.Normal.dist(0, 1) # unregistered rv
  y = a + b * 5
```

In the add logp it would detect the branch on the right `b * 5` is where there is a potential base source for a logo, and that the one on the left will just "shift" it.